prody/proteins/interactions.py

```diff
@@ -0,0 +1,2043 @@
+        if not hasattr(self, '_interactions_matrix') or self._interactions_matrix is None:
+            raise ValueError('Please calculate interactions matrix first')
+
+        import numpy as np
+        import matplotlib
+        import matplotlib.pyplot as plt
+        
+        atoms = self._atoms
+        interaction_matrix = self._interactions_matrix        
+        
+        aa_dic = {'CYS': 'C', 'ASP': 'D', 'SER': 'S', 'GLN': 'Q', 'LYS': 'K',
+             'ILE': 'I', 'PRO': 'P', 'THR': 'T', 'PHE': 'F', 'ASN': 'N', 
+                  'GLY': 'G', 'HIS': 'H', 'LEU': 'L', 'ARG': 'R', 'TRP': 'W', 
+                       'ALA': 'A', 'VAL':'V', 'GLU': 'E', 'TYR': 'Y', 'MET': 'M', 'HSE': 'H', 'HSD': 'H', **kwargs}
+
+        freq_contacts_residues = np.sum(interaction_matrix, axis=0)
+        ResNumb = atoms.select('protein and name CA').getResnums()
+        ResName = atoms.select('protein and name CA').getResnames()
+        ResChid = atoms.select('protein and name CA').getChids()
+        ResList = [ i[0]+str(i[1])+i[2] for i in list(zip(ResName, ResNumb, ResChid)) ]
+
+        all_y = [ aa_dic[i[:3]]+i[3:] for i in  ResList]
+
+        if cutoff > np.max(freq_contacts_residues):
+            cutoff = round(np.max(freq_contacts_residues)*0.7)
+
+        y = []
+        x = []
+        for nr_ii, ii in enumerate(freq_contacts_residues):
+            if ii >= cutoff:
+                x.append(ii)
+                y.append(all_y[nr_ii])
+
+        matplotlib.rcParams['font.size'] = '20' 
+        fig = plt.figure(num=None, figsize=(12,6), facecolor='w')
+        y_pos = np.arange(len(y))
+        
+        plt.bar(y_pos, x, align='center', alpha=0.5, color='blue', **kwargs)
+        plt.xticks(y_pos, y, rotation=45, fontsize=20)
+        plt.ylabel('Score of interactions')
+        plt.tight_layout()
+        plt.show()
+        
+        
+class InteractionsDCD(object):
+
+    """Class for Interaction analysis of DCD trajectory."""
+
+    def __init__(self, name='Unknown'):
+        
+        self._atoms = None
+        self._dcd = None
+        self._interactions_dcd = None
+        self._interactions_nb_dcd = None
+        self._interactions_matrix_dcd = None
+        self._hbs_dcd = None
+        self._sbs_dcd = None
+        self._rib_dcd = None
+        self._piStack_dcd = None
+        self._piCat_dcd = None
+        self._hps_dcd = None
+
+
+    def calcProteinInteractionsDCD(self, atoms, trajectory, **kwargs):
+        """Compute all protein interactions (shown below) for DCD trajectory using default paramaters.
+            (1) Hydrogen bonds
+            (2) Salt Bridges
+            (3) RepulsiveIonicBonding 
+            (4) Pi stacking interactions
+            (5) Pi-cation interactions
+            (6) Hydrophobic interactions
+        
+        :arg atoms: an Atomic object from which residues are selected
+        :type atoms: :class:`.Atomic`
+        
+        :arg trajectory: DCD filename
+        :type trajectory: str
```

Comment:
it could be good to set it like this. At the moment, we are using a Trajectory object and having the problem with getting to the end of the frames